src/mailmap.rs

```diff
@@ -0,0 +1,139 @@
+use std::ffi::CString;
+use std::marker;
+use std::ptr;
+
+use crate::util::Binding;
+use crate::{raw, Error, Repository, Signature};
+
+/// A structure to represent a repository's .mailmap file.
+///
+/// The representation cannot be written to disk.
+pub struct Mailmap<'repo> {
```

Comment:
I'm not sure `'repo` makes sense. I implemented `git_mailmap_from_repository` first and copied this from somewhere else (`Commit`?), however, a `Mailmap` instance can exist independently of a `Repository` as long as `git_mailmap_new` and `git_mailmap_from_buffer` are also implemented.

Reply:
For this you'll probably want to check the implementation on the libgit2 side of things. If `git_mailmap_t` stores pointers to the `git_repository_t` then we'll need the lifetime, otherwise it's safe to remove.

Reply:
[It doesn't](https://github.com/libgit2/libgit2/blob/v1.1.0/src/mailmap.h#L20-L22), I've removed it.